### DISCUSSION TITLE
Disconnect quiet clients after a certain time

### DISCUSSION THREAD

**Discussion by genesisproject2020** [jamulussoftware/jamulus#1116] [Ideas]
I'm running open servers and I see more and more people that connect and just stay connected. I've had clients connected for days without any participation at all. It's fun that people want to listen when people are jamming but I have a hard time to see why some are connected 24/7 without interaction. 

My request is a server function that disconnects clients that have been connected for a long time by them self.

**Comment by ann0see** [comment]
Interesting observation. Probably they are not "real" people more of some kind of "bot"?

I don't think automatically disconnecting a user would solve an issue. They'd just reconnect. Also automatically disconnecting after a time is a bit unfair IMHO.

Quick solution: You could just shut down the server for a few minutes and then bring it up again. This should disconnect the clients.

BUT: this is somehow related to #434 and all the requests around kicking/muting/ and might need to be handled once we think more about this feature.

**Comment by genesisproject2020** [comment]
Your solution of shutting down the server for a few min does not seam to work. I've found that the client just connects again anyway. The solution that work is to block the IP the client uses in the firewall but it's something I like to avoid. 

When thinking about disconnecting... All clients on my servers have to agree on the rules before they are able to get in but I'm wondering if the client get a stream even before answering yes of the agreement.

The #434 is, what I understand, an admin function. I actually think that can be solved with the grouping function so the unwanted sound will not be included. In this scenario I want to avoid clients just connecting and be there without any value. If it's a bot, someone recording or some other strange behaviour, I don't know.

**Comment by dingodoppelt** [comment]
What would be needed is for the admin to have access to the client info connected to his server so that he can use that info to kick or ban clients by passing the IP to the firewall. We are using that on some servers already.

**Comment by genesisproject2020** [comment]
I would hope that a check could be implemented in the server part so if a client have been connected by it self for a certain time the server could block it for some time (or infinite). I've seen clients been connected 12+h by it self.

**Comment by dingodoppelt** [comment]
@genesisproject2020 
since you can't register on jamulus clients would just have to change their names and would be able to reconnect. even blocking by IP means that a client could restart his router to get a new IP. blocking by IP is easy with a firewall and works well. if you really need to block a client I'd go for the firewall solution because restarting the router takes more work than just typing in a new name.

**Comment by gilgongo** [comment]
Hi all - we're trying to make sure Issues are reasonably actionable specifications for work, so if it's OK I'll move this to a discussion until we can raise a ticket or a PR from this. Thanks!
